src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -1509,6 +1570,24 @@ internal SignatureHelp GetSignatureHelp(TextDocumentPosition textDocumentPositio
                 resultCompletionItems = AutoCompleteHelper.GetDefaultCompletionItems(scriptDocumentInfo, useLowerCaseSuggestions);
             }
 
+            //invoke the completion extensions
+            foreach(var completionExt in _completionExtensions.Values)
+            {
```

Comment:
I suspect this code should only run in the code paths that call `GetDefaultCompletionItems`.  In the other cases we are providing schema-based suggestions and my understanding is the model is only providing keyword suggestions.  This implies that these suggestions will most likely not be useful in the other code paths and we probably shouldn't pay the cost to make these calls.

I guess this is a more generic completion provider extensibility mechanism, so ultimately it could make sense to run this all the time.  But until we have a use-case that will provide schema completions from an extension I think the above comment makes sense.

Reply:
This is a generic completion extension point, I don't see the reason to restrict it. And also we are looking at schema-based recommendations as well, we can discuss more offline.

Reply:
If you are concerned about the performance, I can lower the time budget to 50ms.

Reply:
The underlying concern is performance.  The high-level point is if we don't run the `GetDefaultCompletionItems` code path we have already determined from the AST that we are in a location that needs schema-based suggestions.

If we don't have any extensions that provide schema-based suggestions, and don't have any medium-term plans to have such an extension, I would prefer not to pay any cost in this perf-critical code path.  It doesn't really matter if it's 50ms or 200ms if we have very high confidence the suggestions won't be useful.  If we do have a scenario where an extension can provide schema-based suggestions it seems easy enough to relocate this code to run all the time at that time.  

Do see any value in the existing completion extensions running at a place in the code that requires schema-based completions (e.g. in a columns or table list)?

Reply:
...just noted your previous comment "schema-based recommendations as well, we can discuss more offline."  Please ping me to discuss if you plan to support this scenario.  In that case the current integration point would make sense.

Reply:
Based on our offline discussion, we'll limit to default completion list for now.

Reply:
thanks!  looks good to me.  we can switch it back once we have that use-case a little further along.